src/js/components/FormField/FormField.js

```diff
@@ -451,7 +459,47 @@ const FormField = forwardRef(
         {...outerProps}
         style={outerStyle}
         onFocus={(event) => {
-          setFocus(shouldKeepFocus());
+          setFocus(containsFocus(formFieldRef.current));
+          const formDescendants =
+            formFieldRef.current.getElementsByTagName('*');
+          const focusableDescendants = [];
+          for (let i = 0; i < formDescendants.length; i += 1) {
+            const child = formDescendants[i];
+            if (isFocusable(child)) focusableDescendants.push(child);
+
+            let isGrommetInputComponent = false;
+            if (component) {
+              console.log(component.displayName);
+            }
+            if (
+              (children &&
+                children.type &&
+                grommetInputNames.includes(children.type.displayName)) ||
+              (!children &&
+                component &&
+                grommetInputNames.includes(component.displayName)) ||
+              (!children && !component)
+            )
+              isGrommetInputComponent = true;
```

Comment:
I'm not sure if I'm reading this right, but it's trying to decide if the component or child of the formfield is one of the known Grommet input style components but does that mean if the child is a Box that contains a grommet input component it won't notice? And if we add some other grommet input type component we'll have to realize we need to update FormField? It's not clear to me why it really matters if it's a grommet component (ie. that distinction is really complicating this code alot.)

Reply:
If a FormField contains one of the grommet input components, the input components are set up to not have a `focusIndicator` and instead have the focus on the FormField. So for grommet input components even though they have a focusable descendant we don’t want to remove focus from the FormField. For all other components (including custom components) we should handle focus normally and remove focus from the FormField when one of the focusable descendants has focus.

If an input component is wrapped with a Box it won’t be flagged as a grommet input component and focus will be on the FormField. (Tested with Select wrapped in a Box)

As far as updating FormField.js when we add a new grommet input component this is something we would have to do anyway because `grommetInputNames` is being used to determine padding and in other places in FormField.js

Reply:
@jcfilben -- regarding your point re: Select wrapped in Box, are you saying that if I implemented the following, I would still see the double focus on Select search and formfield?

```
<FormField>
   <Box>
      <Select onSearch={() => {}} />
   </Box>
</FormField>
```

Reply:
Sorry I should have wrote that the Select will have focus and not the FormField in this case. If you have Select wrapped in a Box it wouldn't be flagged as a grommet input component and focus would be handled in this statement: `if(focusableDescendants.length >= 1 && !isGrommetInputComponent && focusableDescendants.includes(document.activeElement)) setFocus(false)` (See line 494 of FormField). The select would count as a focusable descendant and it would be the active element when tabbed to so focus on FormField would be set to false as we would see the focusIndicator on Select like this after tabbing to the field:

<img width="367" alt="Screen Shot 2022-01-20 at 1 58 37 PM" src="https://user-images.githubusercontent.com/54560994/150421181-b4661597-2c9a-4f11-95e3-f6a3da2d8ca8.png">

Reply:
Gotcha, thanks for clarifying. Something I'm noticing here (but a separate issue from what you're doing) is the FormField and Select both are having their border now (creating a double border) -- not sure if we have an issue filed around it. It seems to be an effect of only searching the direct children of the formfield

Reply:
The double border shown in the screenshots above is just a result of wrapping Select in a Box within the FormField. If we have `<FormField> <Select /> </FormField>` there is code in the FormField.js file that removes the border of Select so we don't have 2 borders. If you are wrapping Select in a Box within FormField I think the expectation would be that you are doing some custom thing and I think it is reasonable to still show the Select border